lib/puppet-lint/bin.rb

```diff
@@ -76,7 +92,17 @@ def run
           fd.write(l.manifest)
         end
       end
-      puts ']' if PuppetLint.configuration.json
+
+      report_sarif(all_problems, full_base_path, full_base_path_uri) if PuppetLint.configuration.sarif
+
+      if PuppetLint.configuration.json
+        all_problems.each do |problems|
+          problems.each do |problem|
+            [:description, :help_uri].each { |key| problem.delete(key) }
+          end
```

Comment:
There's also `delete_if` which is probably cleaner. Or you could be explicit and use [`Hash.slice`](https://ruby-doc.org/core-2.5.1/Hash.html#method-i-slice) and select the entries that are relevant.

Reply:
yes, changed to use `delete_if`